tasks/src/tasks/select.rs

```diff
@@ -0,0 +1,194 @@
+    flag::{Flag, FlagPerm},
+    mailbox::Mailbox,
+    response::{Code, Data, StatusBody, StatusKind},
+};
+use tracing::{error, warn};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug, Default)]
+pub struct SelectData {
+    // required untagged responses
+    pub flags: Option<Vec<Flag<'static>>>,
+    pub exists: Option<u32>,
+    pub recent: Option<u32>,
+
+    // required OK untagged responses
+    pub unseen: Option<NonZeroU32>,
+    pub permanent_flags: Option<Vec<FlagPerm<'static>>>,
+    pub uid_next: Option<NonZeroU32>,
+    pub uid_validity: Option<NonZeroU32>,
+}
+
+impl SelectData {
+    pub fn validate(self) -> Result<Self, TaskError> {
+        if self.flags.is_none() {
+            warn!("missing required FLAGS untagged response");
+        }
+
+        if self.exists.is_none() {
+            warn!("missing required EXISTS untagged response");
+        }
+
+        if self.recent.is_none() {
+            warn!("missing required RECENT untagged response");
+        }
+
+        if self.unseen.is_none() {
+            warn!("missing required UNSEEN OK untagged response");
+        }
+
+        if self.permanent_flags.is_none() {
+            warn!("missing required PERMANENTFLAGS OK untagged response");
+        }
+
+        if self.uid_next.is_none() {
+            warn!("missing required UIDNEXT OK untagged response");
+        }
+
+        if self.uid_validity.is_none() {
+            warn!("missing required UIDVALIDITY OK untagged response");
+        }
+
+        Ok(self)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SelectTask {
+    mailbox: Mailbox<'static>,
+    read_only: bool,
+    output: SelectData,
+}
+
+impl Task for SelectTask {
+    type Output = Result<SelectData, TaskError>;
+
+    fn command_body(&self) -> CommandBody<'static> {
+        let mailbox = self.mailbox.clone();
+
+        if self.read_only {
+            CommandBody::Examine { mailbox }
+        } else {
+            CommandBody::Select { mailbox }
+        }
+    }
+
+    fn process_data(&mut self, data: Data<'static>) -> Option<Data<'static>> {
+        match data {
+            Data::Flags(flags) => {
+                self.output.flags = Some(flags);
+                None
+            }
+            Data::Exists(count) => {
+                self.output.exists = Some(count);
+                None
+            }
+            Data::Recent(count) => {
+                self.output.recent = Some(count);
+                None
+            }
+            data => Some(data),
+        }
+    }
+
+    fn process_untagged(
+        &mut self,
+        status_body: StatusBody<'static>,
+    ) -> Option<StatusBody<'static>> {
+        let StatusBody { kind, code, text } = status_body;
+
+        match code {
+            Some(Code::Unseen(seq)) => match &kind {
+                StatusKind::Ok => {
+                    self.output.unseen = Some(seq);
+                    None
+                }
+                StatusKind::No => {
```

Comment:
Huch 😕 Didn't think about these cases... I... don't know what to do here. Anything in the RFC? Will try to find out...

Reply:
Okay, I read a bit in the RFC and feel that the most straightforward thing would be to match only exactly what the RFC tells us to match.

For SELECT it says ...

> REQUIRED **OK** untagged responses:  UNSEEN, PERMANENTFLAGS, UIDNEXT, UIDVALIDITY

Thus, I feel like just not processing **NO** and **BAD** and returning as `Some`. These will then show up as unsolicited.